Lib/Src/CController.cpp

```diff
@@ -78,10 +79,11 @@ void CController::run()
 /**
  * @brief Give a new command to controller.
  *
- * @param p_command Pointer to NULL terminated command string.
+ * @param p_command Pointer to a command object that contains command name and
+ * arguments.
```

Comment:
I don't understand this part. I thought ICommand was an interface for the parser. 
`class CJsonParser : public ICommand`. 
But if it is like that, I can't see the point of passing a pointer to a CJsonParser instance. 
The way I see it, it would be easier to pass a `command_t command` retrieved with `getCommand()` from a CJsonParser instance owned by the Dispatcher. 
Let me know if that makes sense.

Reply:
Just to expand a bit, in CDispatcher, the newCommand() method would make a call to the parser. If there is more than one, then a call to all the parsers until a COMMAND_OK status is received. 
```
bool CDispatcher::newCommand(etl::string<MAX_STRING_SIZE> string)
{
    bool b_success = false;
    if(json_parser.parse(string) == COMMAND_OK)
    {
        b_success = true;
    }
   return b_success
}
```
Then, back in the processComChannels method, use getCommand and pass it to all the controllers.